### DISCUSSION TITLE
Why can I use useTranslations hooks from 'next-intl' package on server components?

### DISCUSSION THREAD

**Discussion by shinwonse** [amannn/next-intl#382] [Q&A] [answered]
I am trying to use next.js app router with next-intl package. On next-intl document, they show a example that use hooks on server components.

https://next-intl-docs.vercel.app/docs/environments/server-client-components

I have been known that I can't use hooks on server components, but how can I use useTranslations on server components. I saw next-intl's source code, but they use 'useMemo' and 'useContext' inside useTranslations.

What is different point between normal hooks like 'useState' or 'useEffect' and 'useTranslations' from 'next-intl'?

**Comment by amannn** [comment, marked as answer]
`next-intl` uses [the `react-server` conditional export](https://github.com/reactjs/rfcs/blob/main/text/0227-server-module-conventions.md#react-server-conditional-exports), that's how we can fork the implementation based on the environment of the compilation: https://github.com/amannn/next-intl/blob/feat/next-13-rsc/packages/next-intl/package.json#L24-L26

Hope this helps!
